sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -111,6 +111,10 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
         s"as table property keys may not start with '$DATASOURCE_PREFIX' or '$STATISTICS_PREFIX':" +
         s" ${invalidKeys.mkString("[", ", ", "]")}")
     }
+    // External users are not allowed to set/switch the table type.
+    if (table.properties.contains("EXTERNAL")) {
```

Comment:
I tried Hive. Hive only accepts `EXTERNAL` if users want to change the table type. That means, if users do it like `external` or `ExterRnal`, Hive just treats it as a regular property key.